### ISSUE TITLE
Document BoostConfig.cmake?

### ISSUE THREAD

**Issue by SamB** [boostorg/cmake#77]
The documentation for your version of BoostConfig.cmake seems a little sparse: https://github.com/boostorg/cmake/blob/7f5336b3bf8067bb40da4e8b9940c133271e938a/config/BoostConfig.cmake#L5

CMake 3.30 deprecated [FindBoost](https://cmake.org/cmake/help/latest/module/FindBoost.html), saying:

> Changed in version 3.30: This module is available only if policy [`CMP0167`](https://cmake.org/cmake/help/latest/policy/CMP0167.html#policy:CMP0167) is not set to NEW. Port projects to upstream Boost's `BoostConfig.cmake` package configuration file, for which `find_package(Boost)` now searches.

Sounds reasonable, except they didn't mention that there are two of those, <https://github.com/boostorg/boost_install/blob/develop/BoostConfig.cmake> and <https://github.com/boostorg/cmake/blob/develop/config/BoostConfig.cmake>,  neither of which appears to be fully documented in its own right, seemingly depending on the FindBoost documentation to describe how things are meant to work.

Fortunately, the CMake folks don't seem the type to actually remove deprecated stuff in a hurry, especially not the documentation, but its not exactly clear what the user is actually supposed to before flipping this policy *especially* if they want to make sure their software works with both the "boost_install" and "cmake" variants.

P.S. I sure wish I could open this against *both* repositories. Darn you, github!

**Comment by ulatekh** [comment]
Then how are we supposed to use the new Boost CMake configuration script in a way that's backwards compatible with the old script? My CMake build-system supports a lot of different compilers/versions/platforms simultaneously.

I'm also not thrilled with how the new script makes us do the same thing twice, i.e. list components in the parameters to `find_package(Boost CONFIG)` and then again when linking. A generated value for `Boost_LIBRARIES` was much more handy.

**Comment by pdimov** [comment]
Using targets is the recommended CMake practice (in general, not just for Boost) since CMake 3 ("modern CMake".) FindBoost defines the targets since CMake 3.5. In addition to giving you the include path automatically, targets can also give you compile definitions or compile requirements.

While Boost_LIBRARIES may be slightly more convenient in simple cases, targets are more flexible because you can link to just the necessary targets if you have more than one executable (or library.) E.g.

```
find_package(Boost COMPONENTS lib1 lib2)

add_executable(foo foo.cpp)
target_link_libraries(foo PRIVATE Boost::headers Boost::lib1)

add_executable(bar bar.cpp)
target_link_libraries(bar PRIVATE Boost::headers Boost::lib2)
```

**Comment by ulatekh** [comment]
OK, linking to `Boost::headers` to get the include directory is handy. Was there any existing documentation on that?

And what about the danger of exposing internal details, such as raw target names? My patch above, which sets the `Boost_LIBRARIES` variable with everything that needs to be linked to, based on the most recent call to `find_package(Boost CONFIG ...)`, would be one way to achieve that, plus it helps with backwards compatibility. What do you think?

**Comment by ulatekh** [comment]
Sorry for the late reply... I got caught up reproducing Visual Studio compiler bugs. I'll try this installation procedure as soon as I can.

**Comment by nigels-com** [comment]
@ulatekh Anything further needed for this?
Seems like time to close this ticket.

**Comment by gast128** [comment]
Yes this works. In retrospect the stage directory worked also. So indeed if PREFIX was 'D:\Temp' one needs in CMakeLists.txt:

`set(BOOST_ROOT "D:/Temp")`

or

`set(Boost_DIR "D:/Temp/lib/cmake/Boost-1.86.0")`

Thx. Perhaps somebody could document this in Boost documentation if not already there or in the CMake documentation.
